posthog/management/commands/fix_person_distinct_ids_after_delete.py

```diff
@@ -44,6 +45,10 @@ def run(options, sync: bool = False):
         # it's safer to throw and exit if anything went wrong
         update_distinct_id(distinct_id, version, team_id, live_run, sync)
 
+    logger.info("Waiting on Kafka producer flush, for up to 5 minutes")
+    KafkaProducer().flush(5 * 60)
+    logger.info("Kafka producer queue flushed.")
```

Comment:
Tried some other stuff that I didn't end up particularly happy with and just ended up doing the simple thing and copy/pasting the flush from here: https://github.com/PostHog/posthog/blob/bb50ce52f93f278cf4ecce4a0157838d17ecc1be/posthog/management/commands/split_person.py#L72-L74

Something to note here is that even though `KafkaProducer()` looks a whole lot like a constructor invocation, it's not: https://github.com/PostHog/posthog/blob/bb50ce52f93f278cf4ecce4a0157838d17ecc1be/posthog/kafka_client/client.py#L204

In an ideal world, this would be something that guided you into a path that was more likely to result in an ideal outcome but I don't think it's worth doing a broader refactor of the Kafka-related code right now.

Reply:
Looks good to me. I am not a fan of our singleton pattern usage, but that's a out of scope.